packages/user/Tokens/service/src/lib.rs

```diff
@@ -363,6 +363,27 @@ pub mod service {
         SharedBalance::get_assert(creditor, get_sender(), token_id).debit(amount, memo);
     }
 
+    /// Debit up to
+    ///
+    /// Debits up to X amount of tokens from a shared balance between the creditor and the debitor (sender)
+    ///
+    /// Action will return the amount of tokens actually debited. Set amount to 0 for entire shared balance.
```

Comment:
I don't think using the existing balance to determine the amount to debit is a good idea. The caller of debit should be the primary source of truth of the amount to debit. Making credit primary risks confusion if there is an existing shared balance for any reason.